src/app/hooks/queries/useAccountBalance.ts

```diff
@@ -0,0 +1,170 @@
+
+    let btcBalance = '0';
+    let stxBalance = '0';
+    let ftCoinList: FungibleToken[] | null = null;
+
+    if (account.btcAddress) {
+      const btcData: BtcAddressData = await btcClient.getBalance(account.btcAddress);
+      btcBalance = btcData.finalBalance.toString();
+    }
+
+    if (account.stxAddress) {
+      const apiUrl = `${getNetworkURL(stacksNetwork)}/extended/v1/address/${
+        account.stxAddress
+      }/balances`;
+
+      const response = await axios.get<TokensResponse>(apiUrl, {
+        timeout: API_TIMEOUT_MILLI,
+      });
+
+      const availableBalance = new BigNumber(response.data.stx.balance);
+      const lockedBalance = new BigNumber(response.data.stx.locked);
+      stxBalance = availableBalance.plus(lockedBalance).toString();
+
+      const fungibleTokenList: FungibleToken[] = [];
+      Object.entries(response.data.fungible_tokens).forEach(([key, value]: [string, any]) => {
+        const fungibleToken: FungibleToken = value;
+        const index = key.indexOf('::');
+        fungibleToken.assetName = key.substring(index + 2);
+        fungibleToken.principal = key.substring(0, index);
+        fungibleToken.protocol = 'stacks';
+        fungibleTokenList.push(fungibleToken);
+      });
+
+      const visibleCoins: FungibleToken[] | null = coinsList;
+      if (visibleCoins) {
+        visibleCoins.forEach((visibleCoin) => {
+          const coinToBeUpdated = fungibleTokenList.find(
+            (ft) => ft.principal === visibleCoin.principal,
+          );
+          if (coinToBeUpdated) coinToBeUpdated.visible = visibleCoin.visible;
+          else if (visibleCoin.visible) {
+            visibleCoin.balance = '0';
+            fungibleTokenList.push(visibleCoin);
+          }
+        });
+      } else {
+        fungibleTokenList.forEach((ft) => {
+          ft.visible = true;
+        });
+      }
+
+      const contractids: string[] = fungibleTokenList.map((ft) => ft.principal);
+      const coinsResponse = await getCoinsInfo(network.type, contractids, fiatCurrency);
+
+      if (coinsResponse) {
+        coinsResponse.forEach((coin) => {
+          if (!coin.name) {
+            const coinName = coin.contract.split('.')[1];
+            coin.name = coinName;
+          }
+        });
+
+        // update attributes of fungible token list
+        fungibleTokenList.forEach((ft) => {
+          coinsResponse.forEach((coin) => {
+            if (ft.principal === coin.contract) {
+              ft.ticker = coin.ticker;
+              ft.decimals = coin.decimals;
+              ft.supported = coin.supported;
+              ft.image = coin.image;
+              ft.name = coin.name;
+              ft.tokenFiatRate = coin.tokenFiatRate;
+              coin.visible = ft.visible;
+            }
+          });
+        });
+
+        ftCoinList = fungibleTokenList;
+      }
+    }
+
+    const totalBalance = calculateTotalBalance({
+      stxBalance,
+      btcBalance,
+      ftCoinList,
+      brcCoinsList,
```

Comment:
question: should this be fetched for the processing account instead of the selected one? cause it will sum the same balance to all the accounts right?

Reply:
yeah you're right, I think I overlooked this